Documenting string and bool tensor types + validate bool blob (#842)
RedisAI/RedisAI: docs/commands.md

```

!!! note "Uninitialized Tensor Values"
    As both `BLOB` and `VALUES` are optional arguments, it is possible to use the `AI.TENSORSET` to create an uninitialized tensor (it will contain zeros at all entries).

!!! important "Using `BLOB` is preferable to `VALUES`"
    While it is possible to set the tensor using binary data or numerical values, it is recommended that you use the `BLOB` option. It requires fewer resources and performs better compared to specifying the values discretely.

###Boolean Tensors
The possible values for a tensor of type `BOOL` are `0` and `1`. The size of every bool element in a blob should be 1 byte.   

**Examples**

Here are two ways of creating the following boolean tensor: $\begin{equation*} A = \begin{bmatrix} 0 & 1 \\ 0 & 1 \\ \end{bmatrix} \end{equation*}$
```
redis> AI.TENSORSET my_bool_tensor BOOL 2 2 VALUES 0 1 0 1
OK
redis> AI.TENSORSET my_bool_tensor BOOL 2 2 BLOB "\x00\x01\x00\x01"
OK
```

###String Tensors

> String tensors are tensors in which every element is a single C-string in utf-8 format. A string element can be at any length, and it cannot contain a null-character (unless it is the last character of the string).

Reviewer: ```suggestion
String tensors are tensors in which every element is a single C-string in utf-8 format. A string element can be at any length, and it cannot contain a null-character (unless it is a null-terminated string).
```

> String tensors are tensors in which every element is a single C-string (null-terminated) in utf-8 format. A string element can be at any length, and it cannot contain another null-character except for the terminating one.

Reviewer: ```suggestion
String tensors are tensors in which every element is a single utf-8 string (may or may not be null-terminated). A string element can be at any length, and it cannot contain another null character except for the last one if it is a null-terminated string.
```

> In blob format, string elements are encoded and concatenated, so that the size of string tensor blob is not determined given the tensor's shapes (unlike in the rest of tensor types) 

Reviewer: encoded?
please explain what is the dlimiter.
if the size of a string tensor is not determined given the tensor's shapes, please explain how it is determined
